### PR TITLE
Java: Add HSCAN command

### PR DESCRIPTION
**Pull request by jduo**

### CODE REVIEW

**Comment by jduo** [comment]
This is pointing to @GumpacG 's integration branch for sscan temporarily.

**Comment by jduo** [comment]
The NOVALUES flag was added in Redis 7.2.5 but not marked as such. We will not implement this for now.

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 1]
```diff
@@ -5204,6 +5206,83 @@ public T zscan(@NonNull String key, @NonNull String cursor, @NonNull ZScanOption
...
+     * @see <a href="https://valkey.io/commands/hscan">valkey.io</a> for details.
+     * @param key The key of the hash.
+     * @param cursor The cursor that points to the next iteration of results.
+     * @return Command Response - An <code>Array</code> of <code>Objects</code>. The first element is
+     *     always the <code>cursor</code> for the next iteration of results. <code>0</code> will be
+     *     the <code>cursor</code> returned on the last iteration of the result. The second element is
+     *     always an <code>Array</code> of the subset of the hash held in <code>key</code>. The array
+     *     in the second element is always a flattened series of String pairs, where the key is at
+     *     even indices and the value is at odd indices.
+     * @example
```
Comment (GumpacG) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:5221]: Example does not need to be included in `BaseTransaction`.

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 2]
```diff
@@ -345,6 +346,11 @@ private static Object[] stringCommands(BaseTransaction<?> transaction) {
     private static Object[] hashCommands(BaseTransaction<?> transaction) {
         String hashKey1 = "{HashKey}-1-" + UUID.randomUUID();
 
+        // This extra key is for HScan testing. It is a key with only one field. HScan doesn't guarantee
+        // a
+        // return order but this test compares arrays so order is significant.
```
Comment (GumpacG) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:351]:
```suggestion
        // a return order but this test compares arrays so order is significant.
```

**Review by GumpacG — APPROVED** [review]

**Review by yipin-chen — APPROVED** [review]

File: java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.scan;
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java:2]: make sure to add scan package to [module-info.java](https://github.com/aws/glide-for-redis/blob/main/java/client/src/main/java/module-info.java)

File: java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.List;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * This base class represents the common set of optional arguments for the SCAN family of commands.
+ * Concrete implementations of this class are tied to specific SCAN commands (SCAN, HSCAN, SSCAN,
+ * and ZSCAN).
+ */
+@SuperBuilder
+public abstract class ScanOptions {
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java:14]: would you be able to rename to BaseScanOptions?  As per the description (that already calls it base class). 
We can then add ScanOptions for the Scan command itself.

Comment (jduo) [reply java/client/src/main/java/glide/api/models/commands/scan/ScanOptions.java]: Sure. Did you want to do that here or in the ZScan PRs? Actually, do you want to put this in the same integration branch @acarbonetto

**Comment by GumpacG** [comment]
Closing this as it is no longer necessary. https://github.com/aws/glide-for-redis/pull/1706

**Pull request closed without merging**